### DISCUSSION TITLE
How do I get iterating values in databucket?

### DISCUSSION THREAD

**Discussion by Zcehtro** [mockoon/mockoon#899] [Help] [answered]
I want to use the databucket as a database of sorts to to build objects in different API routes. I have the following databucket:

```javascript
{
  "list": [
    {{#repeat 2}}
    {
      "idMessage": {{@index}},
      "date": "{{date '2022-01-01' '2022-12-12' "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"}}",
      "account": "{{faker 'finance.account'}}",
      "subject": "{{faker 'lorem.words' num: number=3}}",
      "message": "{{faker 'lorem.sentences' 2}}",
      "state": {{faker 'datatype.number' min=2 max=5 }}
    }
    {{/repeat}}
  ]
}
```

In one route, I want to get only the property `idMessage` and `date`. I'm trying with the following:

```javascript
{
  "list": [
    {{#repeat 2}}
    {
      "idMessage": {{data 'messages' 'list.(@index).idMessage'}},
      "date": {{data 'messages' 'list.(@index).date'}}
    }
    {{/repeat}}
  ]
}
```

I'm trying to grab them by trying to parse `@index` but I'm not having success.

What would be proper method to do this?

**Comment by Zcehtro** [comment, marked as answer]
I sort of found the answer myself:

```javascript
{
  "list": [
    {{#each (dataRaw 'messages' 'list')}}
      {
        "idMessage": "{{this.idMessage}}",
        "date": "{{this.date}}"
      }
    {{/each}}
  ]
}
```

But there's still a minor caveat: it doesn't ass the commas automatically like `{{#repeat}}` does. And if I add the comma manually so that it is included in the template, there'll be an unwanted trailling comma at the end of the object.

Trailling comma:

```json
{
  "list": [
    {
      "idMessage": "0",
      "date": "2022-04-23T04:24:08.155Z"
    },
    {
      "idMessage": "1",
      "date": "2022-09-24T01:08:27.659Z"
    },
  ]
}
```

Is there a solution to remove the trailing comma, or a better method than this?

**EDIT**: Solution would be the following:

```javascript
{
  "list": [
    {{#each (dataRaw 'messages' 'list')}}
    {
      "idMessage": {{this.idMessage}},
      "date": "{{this.date}}"
    }{{#if (eq (subtract (len (dataRaw 'messages' 'list')) this.idMessage) 1)}}{{else}},{{/if}}
    {{/each}}
  ]
}
```

Supplied in comment below. If anybody knows a better way, please share.

**Reply by Zcehtro** [reply]
Alright, so I found a solution, albeit not the prettiest one:

```javascript
{
  "list": [
    {{#each (dataRaw 'messages' 'list')}}
    {
      "idMessage": {{this.idMessage}},
      "date": "{{this.date}}"
    }{{#if (eq (subtract (len (dataRaw 'messages' 'list')) this.idMessage) 1)}}{{else}},{{/if}}
    {{/each}}
  ]
}
```

Basically, the {{#if}} takes care of the trailling comma. It's very ugly logic, but it works. If anybody has a proper elegant solution, please share.

**Reply by 255kb** [reply]
You can also use this:
`{{#unless @last}},{{/unless}}`
You have access to the boolean `last` inside an `each`, and there is an unless helper (both part of handlebars).

**Reply by Zcehtro** [reply]
Yes, this is definitely much cleaner. Thank you!
